sdcm/provision/provisioner.py

```diff
@@ -39,9 +40,10 @@ class InstanceDefinition:  # pylint: disable=too-many-instance-attributes
     image_id: str
     type: str   # instance_type from yaml
     user_name: str
-    ssh_key: SSHKey = field(repr=False)
+    ssh_key: SSHKey | str = field(repr=False)
```

Comment:
Can we adapt code that creates `InstanceDefinition` object to provide `SSHKey` object instead of `str`?